### ISSUE TITLE
Low mAP on coco val

### ISSUE THREAD

**Issue by memorywxy** [whai362/PVT#55]
Hello, thx for your work. I was trying to train RetinaNet-FPN-PVTv2-B2-1x model on COCO2017, the reported mAP on val set is 44.6, but the results i got after training was only 33.5. Is there anything wrong?

I trained on 8 V100 GPUs using your provided pre-trained model pvt_v2_b2.pth.
Training script was: 
./dist_train.sh configs/retinanet_pvt_v2_b2_fpn_1x_coco.py 8

The config file  was:
model = dict(
    type='RetinaNet',
    pretrained='/opt/tiger/wanxingyu_tfm/pvt/pretrained/pvt_v2_b2.pth',
    backbone=dict(
        type='pvt_v2_b2',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=True,
        style='pytorch'),
    neck=dict(
        type='FPN',
        in_channels=[64, 128, 320, 512],
        out_channels=256,
        start_level=1,
        add_extra_convs='on_input',
        num_outs=5),
    bbox_head=dict(
        type='RetinaHead',
        num_classes=80,
        in_channels=256,
        stacked_convs=4,
        feat_channels=256,
        anchor_generator=dict(
            type='AnchorGenerator',
            octave_base_scale=4,
            scales_per_octave=3,
            ratios=[0.5, 1.0, 2.0],
            strides=[8, 16, 32, 64, 128]),
        bbox_coder=dict(
            type='DeltaXYWHBBoxCoder',
            target_means=[0.0, 0.0, 0.0, 0.0],
            target_stds=[1.0, 1.0, 1.0, 1.0]),
        loss_cls=dict(
            type='FocalLoss',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_bbox=dict(type='L1Loss', loss_weight=1.0)),
    train_cfg=dict(
        assigner=dict(
            type='MaxIoUAssigner',
            pos_iou_thr=0.5,
            neg_iou_thr=0.4,
            min_pos_iou=0,
            ignore_iof_thr=-1),
        allowed_border=-1,
        pos_weight=-1,
        debug=False),
    test_cfg=dict(
        nms_pre=1000,
        min_bbox_size=0,
        score_thr=0.05,
        nms=dict(type='nms', iou_threshold=0.5),
        max_per_img=100))
dataset_type = 'CocoDataset'
data_root = '/opt/tiger/wanxingyu_tfm/datasets/coco/'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(1333, 800),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type='CocoDataset',
        ann_file=
        '/opt/tiger/wanxingyu_tfm/datasets/coco/annotations/instances_train2017.json',
        img_prefix='/opt/tiger/wanxingyu_tfm/datasets/coco/train2017/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations', with_bbox=True),
            dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
            dict(type='RandomFlip', flip_ratio=0.5),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size_divisor=32),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
        ]),
    val=dict(
        type='CocoDataset',
        ann_file=
        '/opt/tiger/wanxingyu_tfm/datasets/coco/annotations/instances_val2017.json',
        img_prefix='/opt/tiger/wanxingyu_tfm/datasets/coco/val2017/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1333, 800),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='CocoDataset',
        ann_file=
        '/opt/tiger/wanxingyu_tfm/datasets/coco/annotations/instances_val2017.json',
        img_prefix='/opt/tiger/wanxingyu_tfm/datasets/coco/val2017/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1333, 800),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
evaluation = dict(interval=1, metric='bbox')
optimizer = dict(type='AdamW', lr=0.0001, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)
lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=0.001,
    step=[8, 11])
runner = dict(type='EpochBasedRunner', max_epochs=12)
checkpoint_config = dict(interval=1)
log_config = dict(interval=5, hooks=[dict(type='TextLoggerHook')])
custom_hooks = [dict(type='NumClassCheckHook')]
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
work_dir = './work_dirs/retinanet_pvt_v2_b2_fpn_1x_coco'
gpu_ids = range(0, 8)

Test script was:
./dist_test.sh configs/retinanet_pvt_v2_b2_fpn_1x_coco.py work_dirs/retinanet_pvt_v2_b2_fpn_1x_coco/epoch_12.pth 8 --eval bbox

The result i got:
Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.335
Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=1000 ] = 0.514
Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=1000 ] = 0.352
Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.190
Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.356
Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.450
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.525
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=300 ] = 0.525
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=1000 ] = 0.525
Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.325
Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.561
Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.683
OrderedDict([('bbox_mAP', 0.335), ('bbox_mAP_50', 0.514), ('bbox_mAP_75', 0.352), ('bbox_mAP_s', 0.19), ('bbox_mAP_m', 0.356), ('bbox_mAP_l', 0.45), ('bbox_mAP_copypaste', '0.335 0.514 0.352 0.190 0.356 0.450')])

**Comment by whai362** [comment]
It seems the pre-trained model is not loaded correctly.
If the pre-trained model is loaded correctly, you could get a warning like this
![image](https://user-images.githubusercontent.com/8529570/125962716-a71a2aa5-0262-4961-83e2-53755d0182ea.png)

This may cause by the incorrect version of mmdet.
You could install mmdet by this command:
``pip install mmdet==2.13.0 --user''

**Comment by memorywxy** [comment]
Hello, thx for ur reply. I have checked my training log, it seems the pretrained model was not even loaded. 
I already set pretrained='https://github.com/whai362/PVT/releases/download/v2/pvt_v2_b2.pth' in config file, but it didn't download the model when i run dist_train.sh. It seems this line was just ignored, and i got a warning: UserWarning: init_cfg without layer key, if you do not define override key either, this init_cfg will do nothing. However, when i changed the config file to retinanet_pvt_s_fpn_1x_coco_640.py by simply running: bash dist_train.sh configs/retinanet_pvt_s_fpn_1x_coco_640.py 8, it can download the pretrained model correctly. 
Can you help me figure out this problem? BTW, i already installed mmdet 2.13.0 like u said.

**Comment by whai362** [comment]
Hi, you could try to:
1. pull the newest pvt.
2. check the code of loading weights (https://github.com/whai362/PVT/blob/v2/detection/pvt_v2.py#L253).
